synthesis/codon/codon.go

```diff
@@ -149,11 +149,13 @@ func (table *TranslationTable) OptimizeSequence(aminoAcids string, randomState .
 	}
 
 	// weightedRand library insisted setting seed like this. Not sure what environmental side effects exist.
+	var randomSource rand.Source
 	if len(randomState) > 0 {
-		rand.Seed(int64(randomState[0]))
+		randomSource = rand.NewSource(int64(randomState[0]))
```

Comment:
@matiasinsaurralde do you know if there's any change in behavior from the rand.Seed to rand.NewSource migration? This change seems to be breaking codon optimization tests.

Reply:
This wasn't being used properly further down, fixed it in my latest commit.